src/socketio.quiver.js

```diff
@@ -1253,9 +1258,20 @@ QuiverSocialProvider.prototype.onEncryptedMessage_ = function(server, msg) {
     return;
   }
 
-  // TODO: Memoize decryption to save CPU time for identical messages sent
-  // through different paths.
-  this.pgp_.verifyDecrypt(msg.cipherText, msg.key).then(function(plain) {
+  // Memoize decryption.  For security, it's important that msgKey be
+  // non-collidable even if the sender is hostile.
+  var bytes = new Uint8Array(msg.cipherText);
+  var msgKey = msg.key + ';' + Array.prototype.join.call(bytes, ':');
+
+  /** @type {Promise<!FreedomPgpDecryptResult>} */ var plainText;
```

Comment:
Can you call this something like getPlainText or plainTextPromise to emphasize its a promise and not a string?